Test/Unit/Gateway/Request/PaymentDataBuilderTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Alma\MonthlyPayments\Test\Unit\Gateway\Request;
+
+use Alma\MonthlyPayments\Gateway\Config\Config;
+use Alma\MonthlyPayments\Gateway\Config\PaymentPlans\PaymentPlanConfigInterface;
+use Alma\MonthlyPayments\Gateway\Config\PaymentPlans\PaymentPlansConfigInterface;
+use Alma\MonthlyPayments\Gateway\Request\PaymentDataBuilder;
+use Alma\MonthlyPayments\Helpers\ConfigHelper;
+use Magento\Checkout\Model\Session;
+use Magento\Framework\Locale\Resolver;
+use Magento\Payment\Gateway\Data\OrderAdapterInterface;
+use Magento\Payment\Gateway\Data\PaymentDataObject;
+use Magento\Payment\Model\InfoInterface;
+use PHPUnit\Framework\TestCase;
+
+class PaymentDataBuilderTest extends TestCase
+{
+    const RETURN_URL =  'https://mywebsite/alma/payment/return/';
+    const IPN_URL =  'https://mywebsite/alma/payment/ipn/';
+    const CANCEL_URL =  'https://mywebsite/alma/payment/cancel/';
+    const FAILURE_URL =  'https://mywebsite/alma/payment/failure/';
+    const EXPIRATION_TIME =  '2';
+    const INCREMENT_ID =  '100001';
+    const QUOTE_ID =  '454';
+    const PLAN_KEY =  'general:4:0:0';
+    const INSTALLMENT_COUNT =  '4';
+    const LOCALE =  'en_US';
+
+    public function setUp(): void
+    {
+        $this->checkoutSession = $this->createMock(Session::class);
+        $this->config = $this->createMock(Config::class);
+        $this->locale = $this->createMock(Resolver::class);
+        $this->configHelper = $this->createMock(ConfigHelper::class);
+    }
+
+    /**
+     * @dataProvider paymentPayloadDataProvider
+     *
+     * @return void
+     */
+    public function testNonTriggerPaymentPayload($order, $response): void
```

Comment:
```suggestion
    public function testTriggerPaymentPayload($order, $response): void
```

Not only for non-trigger 😉